Zend/zend_builtin_functions.c

```diff
@@ -520,11 +523,10 @@ ZEND_FUNCTION(define)
 				val = &val_free;
 				break;
 			}
```

Comment:
the `if` above seems a bit problematic, since either BC or consistency is harmed, either it is removed, or left as-is.